examples/E2EBert/news_classifier_handler.py

```diff
@@ -121,3 +109,80 @@ def postprocess(self, inference_output):
             return [inference_output]
 
         return inference_output
+
+    def add_attributions_to_visualizer(
+        self,
+        attributions,
+        tokens,
+        pred_prob,
+        pred_class,
+        true_class,
+        attr_class,
+        delta,
+        vis_data_records,
+    ):
+        attributions = attributions.sum(dim=2).squeeze(0)
+        attributions = attributions / torch.norm(attributions)
+        attributions = attributions.detach().numpy()
+
+        # storing couple samples in an array for visualization purposes
+        vis_data_records.append(
+            visualization.VisualizationDataRecord(
+                attributions,
+                pred_prob,
+                pred_class,
+                true_class,
+                attr_class,
+                attributions.sum(),
+                tokens,
+                delta,
+            )
+        )
+
+    def score_func(self, o):
+        output = F.softmax(o, dim=1)
+        pre_pro = np.max(output.detach().numpy())
+        return pre_pro
+
+    def summarize_attributions(self, attributions):
+        """Summarises the attribution across multiple runs
+        Args:
+            attributions ([list): attributions from the Integrated Gradients
+        Returns:
+            list : Returns the attributions after normalizing them.
+        """
+        attributions = attributions.sum(dim=-1).squeeze(0)
+        attributions = attributions / torch.norm(attributions)
+        return attributions
+
+    def explain_handle(self, model_wraper, text, target=1):
+        """Captum explanations handler
+        Args:
+            data_preprocess (Torch Tensor): Preprocessed data to be used for captum
+            raw_data (list): The unprocessed data to get target from the request
+        Returns:
+            dict : A dictionary response with the explanations response."""
+        vis_data_records_base = []
+        model_wrapper = AGNewsmodelWrapper(self.model)
+        tokenizer = BertTokenizer(self.VOCAB_FILE)
+        model_wrapper.eval()
+        model_wrapper.zero_grad()
+        input_ids = torch.tensor([tokenizer.encode(self.text, add_special_tokens=True)])
```

Comment:
@Arvind-Ideas2IT Please move input_ids to device, following it needs changes in Lines #126 and #144/ please run a test on gpu if possible.

Reply:
@HamidShojanazeri  I have updated this as well and made changes in line #126 & #144 as to use  attributions.cpu().detach().numpy()  and np.argmax(output.cpu().detach()) respectively